src/ui/Pager/Pager.ts

```diff
@@ -68,20 +73,35 @@ export class Pager extends Component {
     enableNavigationButton: ComponentOptions.buildBooleanOption({ defaultValue: true }),
 
     /**
-     * Specifies the maximum number of pages to display if enough results are available.
+     * This is a deprecated option. Instead, the pager will automatically adapt itself on each new query.
      *
-     * This property is typically set when the default number of accessible results from the index has been changed from
-     * its default value of `1000` (10 results per page X 100 maxNumberOfPage).
+     * This was used to set the default number of page accessible from the index.
      *
-     * Default value is `100`.
+     * Use {@link Pager.options.maximumNumberOfResultsFromIndex} instead if you modified the settings from the index.
+     *
+     * @deprecated
+     */
+    maxNumberOfPages: ComponentOptions.buildNumberOption({
+      defaultValue: undefined,
+      deprecated: 'This is a deprecated option. The pager will automatically adapt itself on each new query. You no longer need to specify this option. Use maximumNumberOfResultsFromIndex instead.'
+    }),
+    /**
+     * Specify the maximum number of results that the index can return for any query.
+     *
+     * The default value is `1000` in a Coveo index.
+     *
+     * If this value was modified for any reason in the index, you will need to adjust this option so that the Pager component can do it's job correctly.
      */
-    maxNumberOfPages: ComponentOptions.buildNumberOption({ defaultValue: undefined })
+    maximumNumberOfResultsFromIndex: ComponentOptions.buildNumberOption({
```

Comment:
/**
 * Specifies the maximum number of results that the index can return for any query.
 * 
 * Default value is `1000` in a Coveo index.
 *
 * If this value was modified in your Coveo index, you must specify the new value in this option for the `Pager` 
 * component to work properly.
 */